components/eam/src/dynamics/se/se_single_column_mod.F90

```diff
@@ -94,12 +96,23 @@ subroutine scm_setinitial(elem)
               if (have_cldliq) elem(ie)%state%Q(i,j,k,icldliq) = cldliqobs(k)
               if (have_numice) elem(ie)%state%Q(i,j,k,inumice) = numiceobs(k)
               if (have_cldice) elem(ie)%state%Q(i,j,k,icldice) = cldiceobs(k)
-              !  If IOP mode we do NOT want to write over the dy-core vertical 
+              !  If DP-CRM mode we do NOT want to write over the dy-core vertical
               !    velocity with the large-scale one.  wfld is used in forecast.F90
               !    for the compuation of the large-scale subsidence.
-              if (have_omega .and. .not. iop_mode) elem(ie)%derived%omega_p(i,j,k) = wfld(k)
+              if (have_omega .and. .not. dp_crm) elem(ie)%derived%omega_p(i,j,k) = wfld(k)
+              if (dp_crm) elem(ie)%derived%omega_p(i,j,k) = 0.0_real_kind
             enddo
 
+            ! If DP-CRM mode then SHOC/CLUBB needs to know about grid
+            !   length size.  The calculations of this based on a sphere in the
+            !   SHOC and CLUBB interefaces are not valid for a planar grid, thus
+            !   save the grid length from the dycore. Note that planar dycore
+            !   only supports uniform grids, thus we only save one value.
+            if (dp_crm) then
+              ! convert from km to m
+              dyn_dx_size = elem(ie)%dx_short * 1000.0_real_kind
+            endif
+
           endif
```

Comment:
Eventually I will add support for non-uniform planar grids, but this should be fine for now.